File: src/python/_1512_Number_of_Good_Pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        seen = {}
        ans = 0

        for num in nums:
            if num in seen:
                seen[num] += 1
            else:
                seen[num] = 0
            ans += seen[num]

        return ans

File: src/java/_1512_Number_of_Good_Pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        HashMap<Integer, Integer> seen = new HashMap<>();
        int ans = 0;

        for (int num : nums) {
            if (seen.containsKey(num))
                seen.put(num, seen.get(num) + 1);
            else
                seen.put(num, 0);
            ans += seen.get(num);
        }

        return ans;
    }
}
